pkg/kubelet/prober/worker.go

```diff
@@ -203,6 +208,16 @@ func (w *worker) stop() {
 func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 	defer func() { recover() }() // Actually eat panics (HandleCrash takes care of logging)
 	defer runtime.HandleCrash(func(_ interface{}) { keepGoing = true })
+	defer func() {
+		if keepGoing && w.ticker != nil {
+			w.ticker.Reset(w.tickerPeriod)
```

Comment:
should we drain the channel? I'm only confused with the tickers 
https://blogtitle.github.io/go-advanced-concurrency-patterns-part-2-timers/

Reply:
Likely should drain.

If a probe has long processing time and a short period, then the probe's ticker will have triggered queuing another run of doProbe. I talked to Mrunal and he believes that probes are defined as wall time, and the bug lies in the pod definition of the probes.

ref: https://github.com/kubernetes/kubernetes/issues/117153